### PR TITLE
feature: add mstest TestProperty Attributes to test, like Owner or an…

### PR DESCRIPTION
**Pull request by Shakevg**
Added mstest TestProperty Attributes like bellow to RP test Attributes:
[TestCategory(TestCategories.WebClient)]
[Owner(Authors.Developer)]

### CODE REVIEW

File: src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs [hunk 1]
```diff
@@ -219,13 +218,15 @@ private void Events_TestResult(object sender, TestResultEventArgs e)
                         // find categories
                         var testCategories = e.Result.TestCase.Traits.Where(t => t.Name.ToLower() == "Category".ToLower()).Select(x => x.Value).ToList();
 
+                        var attributes = new List<Trait>();
```
Comment (nvborisenko) [inline src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs]: let's rename to `traits`

File: src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs [hunk 2]
```diff
@@ -219,13 +218,15 @@ private void Events_TestResult(object sender, TestResultEventArgs e)
...
                         if (e.Result.TestCase.ExecutorUri.ToString().ToLower().Contains("mstest"))
                         {
                             var testProperty = e.Result.TestCase.Properties.FirstOrDefault(p => p.Id == "MSTestDiscoverer.TestCategory");
                             if (testProperty != null)
                             {
                                 testCategories.AddRange((string[])e.Result.TestCase.GetPropertyValue(testProperty));
                             }
+                            attributes = e.Result.TestCase.Traits.ToList();
```
Comment (nvborisenko) [inline src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs]: for nunit and xunit dot the same, why not. Ideally please test what is stored in `Traits`, I hope not criminal.

File: src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs [hunk 3]
```diff
@@ -219,13 +218,15 @@ private void Events_TestResult(object sender, TestResultEventArgs e)
                         // find categories
                         var testCategories = e.Result.TestCase.Traits.Where(t => t.Name.ToLower() == "Category".ToLower()).Select(x => x.Value).ToList();
```
Comment (nvborisenko) [inline src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs]: you can read all properties here and add into `Attributes`

File: src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs [hunk 4]
```diff
@@ -246,6 +247,10 @@ private void Events_TestResult(object sender, TestResultEventArgs e)
                             Type = TestItemType.Step
                         };
 
+                        // add mstest TestPropertyAttributes
+                        foreach (var itemAttribute in attributes.Select(x => new ItemAttribute
+                                     { Key = x.Name, Value = x.Value })) startTestRequest.Attributes.Add(itemAttribute);
```
Comment (nvborisenko) [inline src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs]: might be `null`

File: src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs (hunk 2 above)
Comment (Shakevg) [reply src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs]: Checked for nunit and xunit too. Nothing extra was added for it.

File: src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs (hunk 4 above)
Comment (Shakevg) [reply src/ReportPortal.VSTest.TestLogger/ReportPortalLogger.cs]: fixed

**Review by nvborisenko — APPROVED** [review]

**Pull request merged**